crates/services/p2p/src/service.rs

```diff
@@ -288,11 +298,14 @@ where
 
                                 let _ = self.p2p_service.send_response_msg(request_id, OutboundResponse::Transactions(transactions_response));
                             }
-                            RequestMessage::SealedHeader(block_height) => {
-                                let response = self.db.get_sealed_header(&block_height)?
-                                    .map(Arc::new);
-
-                                let _ = self.p2p_service.send_response_msg(request_id, OutboundResponse::SealedHeader(response));
+                            RequestMessage::SealedHeaders(range) => {
+                                let max_len = self.max_headers_per_request.try_into().expect("u32 should always fit into usize");
+                                if range.len() > max_len {
+                                    return Err(anyhow!("Requested too many headers: {} > {}", range.len(), max_len));
```

Comment:
It logs the error on the receiver side. It would be much better to return this error to the sender of the request. Otherwise, the sender should wait for the timeout.